### PR TITLE
Fix a couple of memory leaks

### PR DESCRIPTION
**Pull request by themuflon**
Fixes a couple of memory leaks that popped up while investigating with Dr. Memory

### CODE REVIEW

File: loom/engine/loom2d/l2dRectangle.h [hunk 1]
```diff
@@ -260,15 +260,18 @@ class Rectangle
     /**
      * Make a copy of this Rectangle.
      */
-    Rectangle *clone()
+    int clone(lua_State *L)
```
Comment (bengarney) [inline loom/engine/loom2d/l2dRectangle.h]: Why is this the right change? ie what does this accomplish that the normal memory mgmt didn't?

Comment (themuflon) [inline loom/engine/loom2d/l2dRectangle.h]: Not sure, does Lua register the values that are returned?

Comment (SmilyOrg) [inline loom/engine/loom2d/l2dRectangle.h]: Does this actually affect memory usage? I'd be interested in a before/after with just this change. I'm not that familiar with the lua bridge to say either way.

Comment (themuflon) [inline loom/engine/loom2d/l2dRectangle.h]: After testing, I can say yes. After the correct, second change :)

Comment (SmilyOrg) [inline loom/engine/loom2d/l2dRectangle.h]: Cool, can you add a card to check for other places where we might do something similar? E.g. it's possible that the vector code that I wrote does it.

Comment (themuflon) [inline loom/engine/loom2d/l2dRectangle.h]: Will do. I didn't to a lot of searching, but a lot of the code uses the same code, mostly vector stuff :)

Comment (SmilyOrg) [inline loom/engine/loom2d/l2dRectangle.h]: Oops :frowning: :smile:

Comment (themuflon) [inline loom/engine/loom2d/l2dRectangle.h]: Not an oops, that's good

Comment (SmilyOrg) [inline loom/engine/loom2d/l2dRectangle.h]: Oh!

**Comment by SmilyOrg** [comment]
Seems good!

**Pull request merged**